src/transformers/utils/quantizers.py

```diff
@@ -0,0 +1,869 @@
+            A dictionary of remaining keyword arguments with quantization-related kwargs removed.
+        Raises:
+            ValueError: If conflicting arguments are provided or unsupported quantization methods are specified.
+        """
+
+        self.quantization_method = None
+
+        if quantization_config_from_args is not None:
+            self.quantization_config = self.validate_quantization_config(quantization_config_from_args)
+            self.quantization_method = getattr(self.quantization_config, "quant_method")
+
+        elif (kwargs.get("load_in_8bit", False)) or (kwargs.get("load_in_4bit", False)):
+            config_dict = {k: v for k, v in kwargs.items() if k in inspect.signature(BitsAndBytesConfig).parameters}
+            self.quantization_config, kwargs = BitsAndBytesConfig.from_dict(
+                config_dict=config_dict, return_unused_kwargs=True, **kwargs
+            )
+            self.quantization_method = QuantizationMethod.BITS_AND_BYTES
+        else:
+            self.quantization_config = None  # no quantization from args
+
+        if self.quantization_method == QuantizationMethod.BITS_AND_BYTES:
+            bnb_quantization_config_kwargs = {
+                k: v for k, v in kwargs.items() if k in inspect.signature(BitsAndBytesConfig).parameters
+            }
+            if bnb_quantization_config_kwargs:
+                raise ValueError(
+                    "You can't pass `load_in_8bit` or any other `BitsAndBytesConfig` argument as a kwarg when passing "
+                    "`quantization_config` argument at the same time."
+                )
+        # if self.quantization_method == QuantizationMethod.AWQ:
+        #     raise ValueError(
+        #         "You cannot quantize with AWQ a non-quantized model using transformers, please refer to the quantization documentation"
+        #         " to read more about how to quantize models with AWQ algorithm https://huggingface.co/docs/transformers/main_classes/quantization"
+        #     )
+
+        return kwargs
+
+    def validate_quantization_config(self, quantization_config):
+        """
+        checks quantization_config class
+        converts dict format into QuantizationConfigMixin
+        returns quantization_config as QuantizationConfigMixin
+        """
+
+        if isinstance(quantization_config, QuantizationConfigMixin):
+            return quantization_config
+
+        elif isinstance(quantization_config, dict):
+            quant_method = quantization_config.get("quant_method", None)
+
+            if quant_method is None:
+                if quantization_config.get("load_in_8bit", False) or quantization_config.get("load_in_4bit", False):
+                    logger.warning(
+                        "The model's quantization_config has no `quant_method` attribute, but has `load_in_8bit` or `load_in_4bit` parameter set. "
+                        "It is recommended to pass quantization config as `BitsAndBytesConfig`. "
```

Comment:
let's also deprecate these (not twice but either here or in the modeling utils)

Reply:
@younesbelkada suggested to do the same, but in the follow-up PR. 
I had this deprecation notice before, but removed in [this commit](https://github.com/huggingface/transformers/pull/26610/commits/bb5c35a089b6a395ae3502f2dbb97c3e8272eedd)

Reply:
Alright no worries !